marshmallow_jsonapi/fields.py

```diff
@@ -156,9 +176,25 @@ def _serialize(self, value, attr, obj):
             if related_url:
                 ret['links']['related'] = related_url
 
-        if self.include_data:
+        # resource linkage is required when including the data
+        if self.include_resource_linkage or self.include_data:
             if value is None:
                 ret['data'] = [] if self.many else None
             else:
-                ret['data'] = self.add_resource_linkage(value)
+                ret['data'] = self.get_resource_linkage(value)
+
+        if self.include_data and value is not None:
+            if self.many:
+                for item in value:
+                    # items need to be dumped separately,
+                    # othwerwise we'd get a JSONAPI list
```

Comment:
I think it would. You find that clearer? Or maybe that's faster? It's somehow odd to me to first make a jsonapi list just to unpack it afterwards.